rust/src/types.rs

```diff
@@ -168,19 +170,54 @@ impl fmt::Display for RuntimeValueType {
 }
 
 // MistQL runtime value representing all possible data types
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone)]
 pub enum RuntimeValue {
     Null,
     Boolean(bool),
     Number(f64),
     String(String),
-    Object(HashMap<String, RuntimeValue>),
-    Array(Vec<RuntimeValue>),
+    Object(Rc<HashMap<String, Rc<RuntimeValue>>>),
+    Array(Rc<Vec<Rc<RuntimeValue>>>),
     Function(String),
     Regex(MistQLRegex),
 }
 
 impl RuntimeValue {
```

Comment:
NB: Could also likely replace these constructor functions with `pub fn rc(self) -> Rc<Self>` such that we can do `RuntimeValue::Boolean(true).rc()`. Might be slightly clearer / more rust-y / not pollute the returned `RuntimeValue` struct with as many constructors.